apps/hello/templates/services/app/deploy.yml.j2

```diff
@@ -8,6 +8,11 @@ metadata:
   namespace: "{{ project_name }}"
 spec:
   replicas: 1 # number of pods we want
+  selector:
+    matchLabels:
+      app: hello
+      deploymentconfig: "hello-app-{{ deployment_stamp }}"
+      deployment_stamp: "{{ deployment_stamp }}"
```

Comment:
So are the metadata labels below still required? Looks redundant.
If you must keep them replace `deploymentconfig` by `deployment`

Reply:
Yes, both `spec.label.matchLabels` and `spec.template.metadata.labels`  are required and need to be the same.
I'll apply the replacement you suggested.